doc/extensibility/dataflow_example.md

```diff
@@ -0,0 +1,168 @@
+    {
+        base.Initialize();
+        this.broadcastBlock = new BroadcastBlock<string>(
+            null,
+            new DataflowBlockOptions() { NameFormat = "MyOriginalDataSource: {1}" }); // this is recommended to identify your block
+        
+        this.publicBlock = broadcastBlock.SafePublicize(); // CPS extension method to seal off Post
+        this.PostNewValue(this.GetNewValue()); // always publish an initial value!
+
+        // subscribe this.OnChange() to be notified when the original source changes
+        // Could be something like watching for file change events of a config file
+    }
+
+    /// <summary>
+    /// This is called whenever the original source changes
+    /// </summary>
+    private void OnChange()
+    {
+        this.PostNewValue(this.GetNewValue());
+    }
+
+    /// <summary>
+    /// Gets the new value from your original source you are publishing values for
+    /// </summary>
+    private string GetNewValue()
+    {
+        // implementation. Could be reading a file for example.
+    }
+
+    private string PostNewValue(string newValue)
+    {
+        // Add thread safety as needed. Make sure to never regress the data source version published
+        if (newValue != this.lastPublishedValue) // only publish if you have to
+        {
+            this.lastPublishedValue = newValue;
+            this.broadcastBlock.Post(
+                new ProjectVersionedValue<string>(
+                    newValue,
+                    ImmutableDictionary.Create<NamedIdentity, IComparable>().Add(
+                        this.DataSourceKey,
+                        this.sourceVersion++))); // increment your version either before or after publishing. So long as it increments!
+        }
+    }
+}
+```
+
+## Chained Data Source single source
+
+``` Csharp
+/// <summary>
+/// Links to a single upstream source and produces values based on it.
+/// Since this is chained, its versions are derived from the source versions
+/// </summary>
+[Export]
+private class MyChainedDataSource : ChainedProjectValueDataSourceBase<string>
+{
+
+    [ImportingConstructor]
+    protected MyChainedDataSource(UnconfiguredProject project)
+        : base(project.Services)
+    {
+    }
+
+    [Import]
+    private MyOriginalDataSource DataSource { get; set; }
+
+    protected override IDisposable LinkExternalInput(ITargetBlock<IProjectVersionedValue<string>> targetBlock)
+    {
+        // The datasource version is derived from the upstream version,
+        // we do not need a version of our own. However, we must produce a value for every supplied value.
+        // Even if our value does not change.
+        this.JoinUpstreamDataSources(this.DataSource); // Must join upstream to avoid deadlocks!
+        var transformBlock = new TransformBlock<IProjectVersionedValue<string>, IProjectVersionedValue<string>>(
+            i => i.Derive(v => v + "New Value!")); // do some processing with this value and produce a new value
```

Comment:
maybe include a nameFormat as well, so others will follow this pattern to add more debug information to it.?